enablesanitizers.sh

```diff
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
```

Comment:
It is both a build and run-time. I decided to create one .sh to do both, instead of build option + some other .sh or documentation. I can't set the run-time env variables in the build.sh as they wouldn't be present in the environment after the build.sh is finished. The run-time variables used by the sanitizers include ASAN_OPTIONS (which is currently used to turn on run-time memory leak detection)  and ASAN_SYMBOLIZERPATH (which is used to resolve the call stack function names)